rust/sbp/src/codec/fuse.rs

```diff
@@ -0,0 +1,87 @@
+use std::{
+    io::Error,
+    marker::Unpin,
+    ops::{Deref, DerefMut},
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures::io::{AsyncRead, AsyncWrite};
+use pin_project_lite::pin_project;
+
+pin_project! {
+    #[derive(Debug)]
+    pub(crate) struct Fuse<T, U> {
+        #[pin]
+        pub t: T,
+        pub u: U,
+    }
+}
```

Comment:
A comment describing this struct would be helpful. It looks like it's just a wrapper around a Reader/Writer object and an Encoder/Decoder object? It might also be helpful to give the members more descriptive names like `io` and `codec`.